src/Altinn.Profile/Authorization/AuthorizationHelper.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using Altinn.Authorization.ABAC.Xacml.JsonProfile;
+using Altinn.Common.PEP.Authorization;
+using Altinn.Common.PEP.Constants;
+using Altinn.Common.PEP.Helpers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Routing;
+
+namespace Altinn.Profile.Authorization
+{
+    /// <summary>
+    /// Represents a collection of helper methods for creating a decision request
+    /// </summary>
+    public static class AuthorizationHelper
+    {
+        private const string _defaultIssuer = "Altinn";
+        private const string _defaultType = "string";
+        private const string _organizationNumberParam = "organizationNumber";
+
+        /// <summary>
+        /// Creates a decision request based on input
+        /// </summary>
+        /// <returns>Returns a XacmlJsonRequestRoot with data </returns>
+        public static XacmlJsonRequestRoot CreateDecisionRequest(AuthorizationHandlerContext context, ResourceAccessRequirement requirement, RouteData routeData)
+        {
+            XacmlJsonRequest request = new()
+            {
+                AccessSubject = [],
+                Action = [],
+                Resource = []
+            };
+
+            string orgNumber = routeData.Values[_organizationNumberParam] as string;
+
+            request.AccessSubject.Add(DecisionHelper.CreateSubjectCategory(context.User.Claims));
+            request.Action.Add(DecisionHelper.CreateActionCategory(requirement.ActionType));
+
+            if (!string.IsNullOrWhiteSpace(orgNumber))
+            {
+                request.Resource.Add(CreateResourceCategoryForResource(requirement.ResourceId, orgNumber));
+            }
+            else
+            {
+                throw new InvalidOperationException("invalid organization number for this operation: " + orgNumber);
```

Comment:
At this point the orgNumber is NullOrWhiteSpace. Including it in the message is unnecessary.